Software/scan_key/keypad.py

```diff
@@ -33,6 +33,9 @@ def __init__(self, columnCount = 3):
         else:
             return
 
+    def timer(self, sig, frame):
+        raise Exception('Time is up')

```

Comment:
Marking this so I can come back later and review how this Exception modifies execution and how it is logged.